python/cudf/cudf/core/frame.py

```diff
@@ -235,9 +236,26 @@ def _concat(cls, objs, axis=0, ignore_index=False, sort=False):
                     result_index_length += len(obj)
                     empty_has_index = empty_has_index or len(obj) > 0
 
-        # Get a list of the unique table column names
-        names = [name for f in objs for name in f._column_names]
-        names = OrderedDict.fromkeys(names).keys()
+        if join == "inner":
+            index = objs[0].columns if len(objs) > 0 else pd.Index([])
+            for other in objs[1:]:
+                index = index.intersection(other.columns)
```

Comment:
Can we use `objs[0]._column_names` here to avoid having to create a Pandas Index? It's surprisingly expensive to create these Pandas indexes.

Reply:
Sounds good! I will try that out

Reply:
Oh, wait, for this when I use obs[0].columns, is it creating a Pandas Index? I think for the most part it seems like its using the inherent cudf index. Is this correct? I think its only switching to pandas if the series/dataframe is empty. I'll re-write this using the column_names but just checking for future reference. Thank you!

Reply:
Pandas uses its `Index` objects for both row labels and column labels. cuDF has its own `Index` objects for row labels, but since `Column` is a host object (backed by GPU memory), we store column labels on the host. For compatibility reasons, we need `.columns` to return an `Index` object, so we return a Pandas `Index` object, but it's not free to create that Pandas `Index` object.

Reply:
That makes a lot of sense! Thank you!